scipy/optimize/_lsq/lsq_linear.py

```diff
@@ -110,9 +110,9 @@ def lsq_linear(A, b, bounds=(-np.inf, np.inf), method='trf', tol=1e-10,
     fun : ndarray, shape (m,)
         Vector of residuals at the solution.
     optimality : float
-        First-order optimality measure. The uniform norm of the gradient,
-        scaled as described in Notes if the unconstrained solution is not
-        optimal.
+        First-order optimality measure. It is the quantity related to the
+        gradient of the cost function, which was compared with `tol` during
+        iterations.
```

Comment:
This one is not an improvement. "The [I presume you mean a?] quantity related to the gradient" is less precise than the previous explanation.